### PR TITLE
Jira Feature Collector Rewrite

### PR DESCRIPTION
**Pull request by Sbrenthughes**
This PR depends on https://github.com/Hygieia/hygieia-core/pull/46
PR contains refactor of Jira Collector. 
Further testing for users who use `jiraBoardAsTeam=false` will be required.

### CODE REVIEW

File: UI/src/components/widgets/feature/view.js [hunk 1]
```diff
@@ -162,7 +162,7 @@
       if (ctrl.listType === 'issues') {
           ctrl.showFeatureLimitButton = data.result.length <= 4 ? false : true;
       }
-      
+      console.log(data);
```
Comment (nireeshT) [inline UI/src/components/widgets/feature/view.js]: @Sbrenthughes remove logging

File: collectors/feature/jira/src/main/java/com/capitalone/dashboard/model/FeatureCollector.java [hunk 2]
```diff
@@ -12,42 +13,59 @@
...
+
+	public long getLastRefreshTime() {
+		return lastRefreshTime;
+	}
+
+	public void setLastRefreshTime(long lastRefreshTime) {
+		this.lastRefreshTime = lastRefreshTime;
+	}
+
+	public JiraMode getMode() {
```
Comment (nireeshT) [inline collectors/feature/jira/src/main/java/com/capitalone/dashboard/model/FeatureCollector.java]: @Sbrenthughes  jira mode is by default "Board"? how do we set it to "Team" (no setter)

File: collectors/feature/jira/src/main/java/com/capitalone/dashboard/utils/Utilities.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    public static long getLong(JSONObject json, String key) {
+        if (json == null) return 0;
+        Object o = json.get(key);
+        if (o == null) return 0;
+        return (Long) o;
+    }
+
+    //This is weird but way faster than Java date time formatter etc.
+    public static String parseDateWithoutFraction(String date) {
```
Comment (nireeshT) [inline collectors/feature/jira/src/main/java/com/capitalone/dashboard/utils/Utilities.java]: @Sbrenthughes  what is date format in input args ?

File: collectors/feature/jira/src/main/java/com/capitalone/dashboard/model/FeatureCollector.java (hunk 2 above)
Comment (Sbrenthughes) [reply collectors/feature/jira/src/main/java/com/capitalone/dashboard/model/FeatureCollector.java]: it is passed in `FeatureCollectorTask.java` in the `getCollector()` method.

File: collectors/feature/jira/src/main/java/com/capitalone/dashboard/utils/Utilities.java (hunk 3 above)
Comment (Sbrenthughes) [reply collectors/feature/jira/src/main/java/com/capitalone/dashboard/utils/Utilities.java]: Unit tests added.

**Pull request merged**